File: Java/Easy/2022.java
class Solution {
    public int[][] construct2DArray(int[] original, int m, int n) {
        int size = original.length;
        if (size != m * n) {
            return new int[0][0];
        }

        int[][] answer = new int[m][n];

        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                answer[i][j] = original[i * n + j];
            }
        }

        return answer;
    }
}

File: Python/Easy/2022.py
class Solution:
    def construct2DArray(self, original: List[int], m: int, n: int) -> List[List[int]]:
        # check size
        if m * n != len(original):
            return []

        answer = [[0] * n for _ in range(m)]

        for i in range(m):
            for j in range(n):
                answer[i][j] = original[i * n + j]

        return answer
